src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.spec.ts

```diff
@@ -20,11 +56,123 @@ describe('AdminNotificationsPageComponent', () => {
 
   beforeEach(() => {
     fixture = TestBed.createComponent(AdminNotificationsPageComponent);
+    notificationService = TestBed.inject(NotificationService);
+    statusMessageService = TestBed.inject(StatusMessageService);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default view', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when notifications are loading', () => {
+    component.isNotificationLoading = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when notifications failed to load', () => {
+    component.hasNotificationLoadingFailed = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when notification edit form expanded', () => {
+    component.isNotificationLoading = false;
+    fixture.detectChanges();
```

Comment:
This should be moved between lines 90 and 91 so that the changes after clicking the add notification button can be detected

Reply:
After clicking the button, we do not need to call detectChanges() as changes will be inferred from `btn.click()`. But I will add more data for this method to make the snapshot better.

Reply:
`fixture.detechChanges()` is still needed after we modify the attributed and before we click on the button. I will add a `fixture.detectChange()` after clicking the button.